### PR TITLE
added draft typescript definitions

### PR DESCRIPTION
**Pull request by h0ru5**
basically transpiling the webIDL into typescript

### CODE REVIEW

File: typescript-defs/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "keywords": [
+    "web",
+    "WoT",
+    "IoT",
+    "scripting",
+    "typescript",
+    "types"
+  ],
+  "author": "Johannes Hund <johannes.hund@siemens.com>",
+  "license": "MIT",
```
Comment (zolkis) [inline typescript-defs/package.json]: Is MIT license compatible with W3C license?
https://github.com/w3c/wot-scripting-api/blob/master/LICENSE.md

File: typescript-defs/wot.d.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    emitEvent(eventName : string, payload : any) : Promise<void>
+
+    // define request handlers (one per request type, so no events here)
+     onRetrieveProperty(handler : (PropertyRequest) => any) : ExposedThing
+     onUpdateProperty(handler : (PropertyRequest) => any) : ExposedThing
+     onInvokeAction(handler : (ActionRequest) => any) : ExposedThing
+     onObserve(handler : (ObserveRequest) => any) : ExposedThing
+}
+
+interface Request {
```
Comment (zolkis) [inline typescript-defs/wot.d.ts]: Looks good, perhaps we need to update the WebIDL as well.

File: typescript-defs/wot.d.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        name : string;
+        returnValue: any;
+    }
+}
+
+interface ThingDescriptionChangeEvent extends Event {
+    readonly data : {
+        name : string;
+        data: any; // : ThingPropertyInit | ThingActionInit | ThingEventInit 
+        type: TDChangeType;
```
Comment (zolkis) [inline typescript-defs/wot.d.ts]: I think we use `any` when a property can be a simple type or object or `null`, so perhaps we could use the explicit alternatives with union types as marked in the comment, if it doesn't cause any issue. If you have reason to use `any`, we can keep it that way.

Comment (h0ru5) [inline typescript-defs/wot.d.ts]: this was actually a leftover from editing it, the comment was meant to replace the any as soon as the types were defined

File: typescript-defs/package.json (hunk 1 above)
Comment (h0ru5) [inline typescript-defs/package.json]: AFAIK yes, but for the specs it would make sense to (dual-) license it under W3C license

**Comment by zolkis** [comment]
+1 you can merge it :) Thanks!

**Pull request merged**